spec/src/main/asciidoc/configexamples.asciidoc

```diff
@@ -119,3 +119,137 @@ public class InjectedConfigUsageSample {
 }
 ----
 
+=== Aggregate related properties into a single POJO Example
+
+When injecting a number of related configuration properties, it can be tedious to repeat the statement of `ConfigProperty` in scatter places. 
+Since they are related, it makes more sense to aggregate them into a single property class. 
+
+MicroProfile Config provides a way to look up a number of configuration properties starting with the same prefix using the `@ConfigProperties` annotation, e.g. `ConfigProperties(prefix="myPrefix")`.
+When annotating a class with `@ConfigProperties` or `@ConfigProperties(prefix="myPrefix")`, any of its fields, regardless of the visibility, maps to a configuration property via the following mapping rules. 
+
+* If the `prefix` is present, the field `x` maps to the configuration property `<prefix>.x`. 
+* If the `prefix` is absent, the field `x` maps to the property name `x`. 
+
+If the field name `x` needs to be different from the config property name `y`, use `@ConfigProperty(name="y")` to perform the transformation. 
+If the prefix is present, the field `x` maps to the configuration property `<prefix>.y`, otherwise `y`.
+
+Considering the following config sources:
+
+[source, text]
+----
+config_ordinal = 120
+server.host = localhost
+server.port=9080
+server.endpoint=query
+server.old.location=London
+----
+
+[source, text]
+----
+config_ordinal = 150
+client.host = myHost
+client.port=9081
+client.endpoint=shelf
+client.old.location=Dublin
+host = anotherHost
+port=9082
+endpoint=book
+old.location=Berlin
+----
+
+In order to retrieve the above properties in a single property class, you can use the `@ConfigProperties` annotation with a prefix.
+
+[source, java]
+----
+
+@ConfigProperties(prefix="server")
+public class Details {
```

Comment:
You also need to clarify if all fields are visible to set the configuration (not only public fields). I suppose we support everything.

Also, we need to specify that the class requires a no arg constructor.

Reply:
The spec said about the visibility: MicroProfile Config provides a way to look up a number of configuration properties starting with the same prefix using the `@ConfigProperties` annotation, e.g. `ConfigProperties(prefix="myPrefix")`.
When annotating a class with `@ConfigProperties` or `@ConfigProperties(prefix="myPrefix")`, any of its fields, regardless of the visibility, maps to a configuration property via the following mapping rules.

Reply:
Ok, sorry. I guess I've missed that.

What about the constructor part?

Reply:
I thought about the constructor bit. From CDI point of view, in order to be a CDI bean, it needs to have a no-arg constructor or a constructor annotated with `@Inject`. Since we also support non-CDI case, I think it is good to request a zero-arg constructor to avoid instance creation issue.

Reply:
done the changes.